src/setfit/modeling.py

```diff
@@ -305,13 +297,33 @@ def _freeze_or_not(self, model: torch.nn.Module, to_freeze: bool) -> None:
         for param in model.parameters():
             param.requires_grad = not to_freeze
 
-    def predict(self, x_test: torch.Tensor) -> torch.Tensor:
-        embeddings = self.model_body.encode(x_test, normalize_embeddings=self.normalize_embeddings)
-        return self.model_head.predict(embeddings)
+    def predict(self, x_test: List[str], as_numpy: bool = False) -> Union[torch.Tensor, "ndarray"]:
+        embeddings = self.model_body.encode(
+            x_test, normalize_embeddings=self.normalize_embeddings, convert_to_tensor=True
+        )
 
-    def predict_proba(self, x_test: torch.Tensor) -> torch.Tensor:
-        embeddings = self.model_body.encode(x_test, normalize_embeddings=self.normalize_embeddings)
-        return self.model_head.predict_proba(embeddings)
+        out = self.model_head.predict(embeddings)
```

Comment:
nit: we usually aim for explicit variable names where possible :)

```suggestion
        outputs = self.model_head.predict(embeddings)
```